File: python/1. Two Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        n = len(nums)
        for i in range(n):
            for j in range(i+1, n):
                if nums[i] + nums[j] == target:
                    return [i, j]

if __name__ == '__main__':
    nums = [3,2,4]
    target = 6
    su = Solution()
    print su.twoSum(nums, target)                    
    foo = lambda x: x>1
    print foo(1)

File: go/src/leetcode/1. Two Sum/1. Two Sum.go
package main

import "fmt"

func twoSum(nums []int, target int) []int {
	mem := make(map[int]int)
	for i, num := range nums {
		other := target - num
		if _, ok := mem[other]; ok {
			return []int{mem[other], i}
		}
		mem[num] = i
	}
	return []int{-1, -1}
}

func main() {
	nums := []int{2,7,11,15}
	target := 9
	fmt.Println(twoSum(nums, target))
}
